### ISSUE TITLE
Need to be corrected for '\'

### ISSUE THREAD

**Issue by vipasko** [koalaman/shellcheck#2843]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1003
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo '\' # good
echo "\"" # good too
```

#### Here's what shellcheck currently says:
```
Line 2	SC1003: Want to escape a single quote? echo 'This is how it'\''s done'.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output
